### PR TITLE
Add Fletching

### PR DESCRIPTION
**Pull request by coolbop32**
### Description:

-   Adds the Fletching Skill

### Changes:

-   _Add sinew to cooking_
    _Add crossbow string to crafting_
    _Made feather pack give more and closer to the in-game price_

-   [x] I have tested all my changes thoroughly.

### CODE REVIEW

File: src/commands/Minion/fletch.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+			cooldown: 1,
+			usage: '[quantity:int{1}|name:...string] [name:...string]',
+			usageDelim: ' '
+		});
+	}
+
+	async run(msg: KlasaMessage, [quantity, fletchName = '']: [null | number | string, string]) {
+		if (!msg.author.hasMinion) {
+			throw `You dont have a minion`;
+		}
```
Comment (gc) [inline src/commands/Minion/fletch.ts:33]: use the decorators for this, and in any commands youve added.

File: src/commands/Minion/fletch.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+								.join(', ')}`
+					).join('\n')
+				),
+				`Available fletching items.txt`
+			);
+		}
+
+		if (msg.author.minionIsBusy) {
+			return msg.send(msg.author.minionStatus);
+		}
```
Comment (gc) [inline src/commands/Minion/fletch.ts:51]: this should be removed in favour of the decorator, as the above comment says.

File: src/commands/Minion/fletch.ts [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+		if (typeof quantity === 'string') {
+			fletchName = quantity;
+			quantity = null;
+		}
+
+		const Fletch = Fletching.Fletchables.find(item => stringMatches(item.name, fletchName));
```
Comment (gc) [inline src/commands/Minion/fletch.ts:58]: this variable should be called `fletchableItem` or `itemToFletch` or something like that.

File: src/extendables/User/Minion.ts [hunk 4]
```diff
@@ -58,6 +60,7 @@ export default class extends Extendable {
 - Train woodcutting with \`+chop\`
 - Train firemaking with \`+light\`
 - Train crafting with \`+craft\`
+- Train dletching with \`+fletch\`
```
Comment (gc) [inline src/extendables/User/Minion.ts:63]: fletching

File: src/lib/createables.ts [hunk 5]
```diff
@@ -675,6 +676,26 @@ const Createables: Createable[] = [
...
+			[itemID(`Zulrah's Scales`)]: 25000
+		}
+	},
+	{
+		name: 'Toxic blowpipe',
+		inputItems: {
+			[itemID('Toxic blowpipe (empty)')]: 1,
+			[itemID(`Zulrah's Scales`)]: 25000
+		},
+		outputItems: {
```
Comment (gc) [inline src/lib/createables.ts:696]: i think we should remove this for now... it wont have a use immediately anyway, and i need to decide if ill code an actual blowpipe (and ammo) system for consuming amo

File: src/tasks/minions/fletchingActivity.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+import { channelIsSendable } from '../../lib/util/channelIsSendable';
+import { FletchingActivityTaskOptions } from '../../lib/types/minions';
+import Fletching from '../../lib/skilling/skills/fletching/fletching';
+
+export default class extends Task {
+	async run({ fletchableID, quantity, userID, channelID }: FletchingActivityTaskOptions) {
+		const user = await this.client.users.fetch(userID);
+		const currentLevel = user.skillLevel(SkillsEnum.Fletching);
+
+		const Fletchable = Fletching.Fletchables.find(fletchable => fletchable.id === fletchableID);
```
Comment (gc) [inline src/tasks/minions/fletchingActivity.ts:16]: same variable name changing.

File: src/commands/Minion/fletch.ts [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+			duration,
+			type: Activity.Fletching,
+			id: rand(1, 10_000_000),
+			finishDate: Date.now() + duration
+		};
+
+		// Remove the required items from their bank.
+		let newBank = { ...userBank };
+		for (const [itemID, qty] of requiredItems) {
+			if (parseInt(itemID) === 995) {
```
Comment (Alexsuperfly) [inline src/commands/Minion/fletch.ts:123]: no coins are ever in required items

File: src/tasks/minions/fletchingActivity.ts [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+		const fletchableItem = Fletching.Fletchables.find(
+			fletchable => fletchable.id === fletchableID
+		);
+
+		if (!fletchableItem) return;
+
+		const xpReceived = quantity * fletchableItem.xp;
+		let newQuantity;
+		if (!fletchableItem.outputMultiple) {
```
Comment (Alexsuperfly) [inline src/tasks/minions/fletchingActivity.ts:24]: you should be able to just *= quantity by multiple if its set

File: src/lib/skilling/skills/fletching/fletchables/arrows.ts [hunk 9]
```diff
@@ -0,0 +1,87 @@
+import itemID from '../../../../util/itemID';
+import { Fletchable } from '../../../types';
+
+const Arrows: Fletchable[] = [
+	{
+		name: 'Headless arrow',
+		id: itemID('Headless arrow'),
+		level: 1,
+		xp: 1,
+		inputItems: { [itemID('Arrow shaft')]: 1, [itemID('Feather')]: 1 },
```
Comment (Alexsuperfly) [inline src/lib/skilling/skills/fletching/fletchables/arrows.ts:10]: i think transforming all of these to [this](https://github.com/gc/oldschoolbot/blob/0fce12db1a7fd72a4dd728a067f49b198ac92054/src/lib/skilling/skills/crafting/craftables/dragonhide.ts#L99) format would make it more readable too

**Pull request merged**